client/src/partials/cost_center/center/analysis_center.js

```diff
@@ -1,122 +1,111 @@
+    query : {
+      tables : {
+        'cost_center' : {
+          columns : ['id', 'text', 'note', 'is_principal']
         }
       }
-    };
-
-    $scope.register = {};
-    $scope.selected = {};
-
-    startup();
-
-    function startup() {
-      $scope.enterprise = SessionService.enterprise;
-      validate.process(dependencies).then(init);
-    }
-
-    function init(model) {
-      $scope.model = model;
-    }
-
-    function setAction(value, cost_center) {
-      $scope.action = value;
-      $scope.selected = angular.copy(cost_center) || {};
-      $scope.selected.is_principal = $scope.selected.is_principal !== 0;
     }
+  };
 
-    function writeCenter() {
-      return connect.post('cost_center', connect.clean($scope.register));
-    }
+  vm.register = {};
+  vm.selected = {};
 
-    function saveRegistration() {
-      $scope.register.is_principal = ($scope.register.is_principal)? 1 : 0;
-      writeCenter()
-      .then(function() {
-        validate.refresh(dependencies, ['cost_centers']).then(function (model) {
-          angular.extend($scope, model);
-        });
+  startup();
 
-        $scope.register = {};
+  function startup() {
+    session.state = 'loading';
+    vm.enterprise = SessionService.enterprise;
+    validate.process(dependencies).then(init);
+  }
 
-        messenger.success($translate.instant('ANALYSIS_CENTER.INSERT_SUCCESS_MESSAGE'));
-      })
-      .catch(function () {
-        messenger.danger($translate.instant('ANALYSIS_CENTER.INSERT_FAIL_MESSAGE'));
-      });
-    }
+  function init(model) {
+    vm.model = model;
+    session.state = 'loaded';
+  }
 
-    /*
-    function getAvailablesAccounts(enterprise_id) {
-      return connect.fetch('/availablechargeAccounts/'+$scope.project.enterprise_id+'/');
-    }
+  function setAction(value, cost_center) {
+    vm.action = value;
+    vm.selected = angular.copy(cost_center) || {};
+    vm.selected.is_principal = vm.selected.is_principal !== 0;
+  }
 
-    function loadAssociateAccounts() {
-      return connect.fetch('/costCenterAccount/' + $scope.proejct.enterprise_id + '/' + $scope.selected.id);
-    }
+  function writeCenter() {
+    return connect.post('cost_center', connect.clean(vm.register));
+  }
 
-    function handleConfigure() {
-      loadAssociateAccounts()
-      .then(function (values) {
-        models.associatedAccounts = values;
-        transformDatas(models.associatedAccounts);
-      });
-    }
-    */
-
-    function remove(cost_center) {
-      $scope.selected = angular.copy(cost_center);
-      removeCostcenter()
-      .then(function () {
-        $scope.model.cost_centers.remove($scope.selected.id);
-        messenger.success($translate.instant('ANALYSIS_CENTER.REMOVE_SUCCESS_MESSAGE'));
-      })
-      .catch(function () {
-        messenger.danger($translate.instant('ANALYSIS_CENTER.REMOVE_FAIL_MESSAGE'));
+  function save() {
+    vm.register.is_principal = (vm.register.is_principal)? 1 : 0;
```

Comment:
See [this comment](https://github.com/IMA-WorldHealth/bhima/pull/799/files#r41849772).